File: src/main/java/com/fionera/hadoop/RelaxReducer.java
package com.fionera.hadoop;

import org.apache.hadoop.io.Text;
import org.apache.hadoop.mapreduce.Reducer;

import java.io.IOException;

public class RelaxReducer
        extends Reducer<Text, Text, Text, Text> {

    @Override
    protected void reduce(Text key, Iterable<Text> values,
                          Context context) throws IOException, InterruptedException {
        StringBuilder list = new StringBuilder();
        for (Text value : values) {
            list.append(value).append(" ");
        }
        context.write(key, new Text(list.toString()));
    }
}

File: src/main/kotlin/com/fionera/hadoop/mapred/reducer/RelaxReducer.kt
package com.fionera.hadoop.mapred.reducer

import org.apache.hadoop.io.Text
import org.apache.hadoop.mapreduce.Reducer

import java.io.IOException

class RelaxReducer : Reducer<Text, Text, Text, Text>() {

    @Throws(IOException::class, InterruptedException::class)
    override fun reduce(key: Text, values: Iterable<Text>,
                        context: Context) {
        val list = StringBuilder()
        for (value in values) {
            list.append(value).append(" ")
        }
        context.write(key, Text(list.toString()))
    }
}
